src/tsquery-types.ts

```diff
@@ -8,6 +8,7 @@ export type TSQueryApi = {
    matches (node: TSQueryNode, selector: TSQuerySelectorNode, ancestry: Array<TSQueryNode>): boolean;
    parse (selector: string): TSQuerySelectorNode;
    query <T extends Node = Node> (ast: Node | TSQueryNode<T>, selector: string): Array<TSQueryNode<T>>;
+   syntaxKindName (node: SyntaxKind): string;
```

Comment:
Do we need to make this public? What's the usecase?

Reply:
This way TSQuery Playground and the VSCode extension will be able to reuse the same logic to display the correct syntax kind for each node

Reply:
Wouldn't they just use `node.kindName`?

Reply:
Nope, you only "enrich" the nodes when running a query. When we display the AST in the side view we don't actually run a query. With VSCode, there are also performance considerations, since this runs whenever you change anything in your code

Reply:
Does the hack in `getPath` in `utils.ts` fix that?

Reply:
It could, but I think an API that would simply return the new properties (instead of modifying the original object) would be better - both in terms of performance and not breaking future typescript if they change the behavior of some of these properties

Reply:
You could easily rewrite util.ts to use that API:

instead of 
```ts
addProperties(obj as TSQueryNode);
```

Just do:
```ts
Object.assign(obj, tsqueryProperties(obj));
```

where tsqueryProperties will simply return the new properties instead of assigning them to `obj`

Reply:
Aight cool, let's go with this for now, could you resolve those conflicts?

Reply:
sure, just a moment

Reply:
here we go!